### DISCUSSION TITLE
How do we send the authentication request?

### DISCUSSION THREAD

**Discussion by ADParris** [gladly-team/next-firebase-auth#219] [Q&A]
I've read over this page about ten times, over the past several hours, so yeah... I've also read over the docs for 'firebaseui' as that seems to be how you are doing the pitching, as this library appears to only be a catcher. I guess my question is, say someone comes to the site and wants to log in via an email and password... How do we send that request, with this package? Just use the firebase auth as I normally would with the whole 'signInWithEmailAndPassword' thing? And, if so, how does that get connected to this component? It just magically catches the return auth conversation, with the Firebase Servers? Not really understanding the entire flow here, as you can see. Any insight would be appreciated!

**Comment by kmjennison** [comment]
What you're describing is handled automatically. This package listens to the Firebase JS SDK for auth change events and makes the calls to your login/logout endpoints. See [`useFirebaseUser`](https://github.com/gladly-team/next-firebase-auth/blob/main/src/useFirebaseUser.js#L82) for this logic in action.

As for actually signing in via Firebase, you can do this however you'd like. `firebaseui-web` is one option.

**Reply by ADParris** [reply]
Okay, thank you. That is what I had gathered, just wanted to clarify it and due to you using the 'firebaseui' in your code further masking what was going on ( which is fine normally, by the way, I actually didn't know about that package and had been writing all of the logic out by hand ) it was a bit confusing is all. It's fantastic actually, between 'firebaseui' and this package firebase truly becomes 'drop in and go'. I got it working earlier this morning. Though I did have to override some of their styles.  Anyway, again thanks for the clarification!  🙂
